include/openssl/asn1t.h

```diff
@@ -42,7 +42,7 @@ extern "C" {
                 static const ASN1_ITEM local_it = {
 
 # define static_ASN1_ITEM_start(itname) \
-        static ASN1_ITEM_start(itname)
+        static ossl_unused ASN1_ITEM_start(itname)
```

Comment:
This is a macro that actually a macro that expands to being a variable

```
const ASN1_ITEM * X509_ALGORS_it(void)
{
    ``static const ASN1_ITEM local_it`` = {
                0x0,
                -1,
                &X509_ALGORS_item_tt,
                0,
                ((void *)0),
                0,
                "X509_ALGORS"
        };
        return &local_it;
}
```

Reply:
Is this really an unused-function ??

Reply:
I think the full expansion is that it defines a static function that internally defines and returns a static variable. Perhaps I am reading it incorrectly.

The function is almost certainly not globally unused within OpenSSL, but the thinking was that it might not be used within the same header and it would therefore be possible for some translation unit not to use it. It was kind of hard to suss this out with the functions buried in macros like this, but since there were very few such cases I decided to err on the side of caution with it; I don't really figure ossl_unused is doing any harm even if it may not be necessary. If you can show that the macro-generated function here will always be used in the same header in which it's defined, I would of course be happy to drop the change.